From 200eca7f1419b1ae53958b51e8551f7e7f6cd467 Mon Sep 17 00:00:00 2001
From: Marcelo Ricardo Leitner <marcelo.leitner@gmail.com>
Date: Tue, 30 Apr 2019 11:31:59 -0300
Subject: [PATCH] automake: fix include dir for the now autogenerated header

After 9607dd85e70a ("netinet/sctp.h: dynamically build based on system
setup") the header file is generated on the builddir and won't be
located in the srcdir anymore. This broke builds using different dirs
for building other than the src dir.

Fix it by telling automake to use the include dir based on top_builddir
instead.

Fixes #30

Fixes: 9607dd85e70a ("netinet/sctp.h: dynamically build based on system setup")
Reported-by: Alexander Gallego
Acked-by: Neil Horman <nhorman@tuxdriver.com>
Signed-off-by: Marcelo Ricardo Leitner <marcelo.leitner@gmail.com>
Signed-off-by: Srivatsa S. Bhat (VMware) <srivatsa@csail.mit.edu>
---
 src/apps/Makefile.am       | 2 +-
 src/func_tests/Makefile.am | 2 +-
 src/lib/Makefile.am        | 2 +-
 src/testlib/Makefile.am    | 2 +-
 src/withsctp/Makefile.am   | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/apps/Makefile.am b/src/apps/Makefile.am
index 7e33b9c..7e32306 100644
--- a/src/apps/Makefile.am
+++ b/src/apps/Makefile.am
@@ -4,7 +4,7 @@ include $(top_srcdir)/Makefile.rules
 include $(top_srcdir)/Makefile.dirs
 
 # General compilation flags
-AM_CPPFLAGS = -I. -I$(top_srcdir)/src/include -I$(top_srcdir)/src/testlib \
+AM_CPPFLAGS = -I. -I$(top_builddir)/src/include -I$(top_srcdir)/src/testlib \
               -g -O2 -fno-strict-aliasing -Wall -Wstrict-prototypes \
               -Wimplicit-function-declaration
 
diff --git a/src/func_tests/Makefile.am b/src/func_tests/Makefile.am
index 495adcf..e5bf454 100644
--- a/src/func_tests/Makefile.am
+++ b/src/func_tests/Makefile.am
@@ -4,7 +4,7 @@ include $(top_srcdir)/Makefile.rules
 include $(top_srcdir)/Makefile.dirs
 
 # General compilation flags
-AM_CPPFLAGS = -I. -I$(top_srcdir)/src/include -I$(top_srcdir)/src/testlib \
+AM_CPPFLAGS = -I. -I$(top_builddir)/src/include -I$(top_srcdir)/src/testlib \
               -g -O2 -fno-strict-aliasing -Wall -Wstrict-prototypes \
               -Wimplicit-function-declaration
 
diff --git a/src/lib/Makefile.am b/src/lib/Makefile.am
index 1d62175..1296caa 100644
--- a/src/lib/Makefile.am
+++ b/src/lib/Makefile.am
@@ -4,7 +4,7 @@ include $(top_srcdir)/Makefile.rules
 include $(top_srcdir)/Makefile.dirs
 
 # General compilation flags
-AM_CPPFLAGS = -I$(top_srcdir)/src/include
+AM_CPPFLAGS = -I$(top_builddir)/src/include
 
 lib_LTLIBRARIES = libsctp.la
 
diff --git a/src/testlib/Makefile.am b/src/testlib/Makefile.am
index fae6bbc..b949611 100644
--- a/src/testlib/Makefile.am
+++ b/src/testlib/Makefile.am
@@ -5,7 +5,7 @@ include $(top_srcdir)/Makefile.rules
 include $(top_srcdir)/Makefile.dirs
 
 # General compilation flags
-AM_CPPFLAGS = -I$(top_srcdir)/src/include
+AM_CPPFLAGS = -I$(top_builddir)/src/include
 
 noinst_LTLIBRARIES = libsctputil.la
 libsctputil_la_SOURCES = sctputil.c sctputil.h
diff --git a/src/withsctp/Makefile.am b/src/withsctp/Makefile.am
index 1f6ca37..3157588 100644
--- a/src/withsctp/Makefile.am
+++ b/src/withsctp/Makefile.am
@@ -10,7 +10,7 @@ include $(top_srcdir)/Makefile.rules
 bin_PROGRAMS = checksctp
 bin_SCRIPTS = withsctp
 
-AM_CPPFLAGS=-I$(top_srcdir)/src/include
+AM_CPPFLAGS=-I$(top_builddir)/src/include
 pkglib_LTLIBRARIES = libwithsctp.la
 libwithsctp_la_SOURCES = sctp_load_libs.c sctp_socket.c sctp_bind.c \
 	sctp_sockopt.c sctp_socket.h
